src/qat/core.py

```diff
@@ -6,31 +6,107 @@
 
 from qat.compiler.analysis_passes import InputAnalysis
 from qat.compiler.transform_passes import InputOptimisation, InputOptimisationResult, Parse
-from qat.ir.pass_base import PassManager, QatIR
+from qat.ir.pass_base import PassManager, QatIR, TransformPass
 from qat.ir.result_base import ResultManager
 from qat.purr.backends.realtime_chip_simulator import get_default_RTCS_hardware
+from qat.purr.backends.echo import get_default_echo_hardware
 from qat.purr.compiler.builders import InstructionBuilder
 from qat.purr.compiler.frontends import QASMFrontend
 from qat.purr.compiler.hardware_models import QuantumHardwareModel
 from qat.purr.compiler.metrics import CompilationMetrics
 from qat.purr.qatconfig import QatConfig
 from qat.qat import QATInput
 
+from qat.pipelines import DefaultCompile, DefaultExecute
+from qat.coreconfig import PipelinesConfig
+
 
 class QAT:
-    def __init__(self, hardware_model: Optional[QuantumHardwareModel] = None):
-        self.config = QatConfig()
-        self.hardware_model = hardware_model
+    def __init__(
+        self,
+        qatconfig: Optional[QatConfig] = None,
+        pipelinesconfig: Optional[PipelinesConfig] = None,
+    ):
+
+        self._compile_pipelines = {}
+        self._execute_pipelines = {}
```

Comment:
`PassManager` is already a container of leaf passes or composite passes themselves, you could reuse here:

```
self._pipelines = PassManager()

# ...

compile_pipelines = PassManager() # A pass but also a pipeline
# add leaf compile passes

execute_pipelines = PassManager() # A pass but also a pipeline
# add leaf execute passes

self._pipelines.add(compile_pipelines)
self._pipelines.add(execute_pipelines)
```